python/dgl/data/movielens.py

```diff
@@ -0,0 +1,630 @@
+"""MovieLens dataset"""
+
+import os
+import re
+import zipfile
+
+import numpy as np
+import pandas as pd
+import torch as th
+from sklearn.model_selection import train_test_split
+
+import dgl
+from dgl.data import DGLDataset
+
+from dgl.data.utils import (
+    download,
+    load_graphs,
+    load_info,
+    save_graphs,
+    save_info,
+)
```

Comment:
Use relative imports to be consistent with the current DGL practice.

Reply:
This issue and the one below are solved in the latest commit. There is one point. I was trying to make movielens support multiple platforms. However, I use torchtext to preprocess text embeddings following GCMC practice. This may lead to movielens depending on PyTorch since torchtext might depend on PyTorch.

Reply:
Another question is, I'm not quite familiar with unit test of dgl if we need to test movielens (I write a test script for movielens following actor dataset PR. But I didn't know how to test it). Could you guide me to some documents of the unit test pipeline in dgl?

Reply:
For the first issue, it's fine to only support PyTorch. You can follow the practice in https://github.com/dmlc/dgl/blob/master/python/dgl/homophily.py

Reply:
For a local test, you can simply build DGL from source and test the code snippet. You don't necessarily need to launch the unit test pipeline locally.

Reply:
> For the first issue, it's fine to only support PyTorch. You can follow the practice in https://github.com/dmlc/dgl/blob/master/python/dgl/homophily.py

In the lastest commit, I've already follow the practice in the link to update PyTorch support.

Reply:
> For a local test, you can simply build DGL from source and test the code snippet. You don't necessarily need to launch the unit test pipeline locally.

I've already built DGL from source and briefly test MovieLens. Seems MovieLens work well. 
```python
import dgl

print(dgl.__version__)

from dgl.data import MovieLensDataset

dataset = MovieLensDataset('ml-100k', valid_ratio=0.2, test_ratio=0.1)
train_g, valid_g, test_g = dataset[0]
info, feat = dataset.info, dataset.feat

train_ratings, valid_ratings, test_ratings = \
    info['train_rating_pairs'], info['valid_rating_pairs'], info['test_rating_pairs']
user_feat, movie_feat = \
    feat['user_feat'], feat['movie_feat']

pass
# I have tested the shape of variables above, and some statistics of graphs, 
# such as number of nodes and edges. No error.
```